src/test/java/org/logstash/beats/BeatsParserTest.java

```diff
@@ -50,14 +53,25 @@ public void testEncodingDecodingFields() {
         assertMessages(this.batch, decodedBatch);
     }
 
-
     @Test
-    public void  testCompressedEncodingDecodingFields() {
+    public void testCompressedEncodingDecodingFields() {
         this.batch.setProtocol(Protocol.VERSION_1);
         Batch decodedBatch = decodeCompressedBatch();
         assertMessages(this.batch, decodedBatch);
     }
 
+    @Test
+    public void testShouldNotCrashOnGarbageData() {
```

Comment:
Yes, I will do that, I am actually leaning toward removing the exception caught in the BeatsParser and add a new handler to only deal with that and would make code easier to test?